integrations/object_store/src/lib.rs

```diff
@@ -62,12 +65,34 @@ impl std::fmt::Display for OpendalStore {
 
 #[async_trait]
 impl ObjectStore for OpendalStore {
-    async fn put(&self, location: &Path, bytes: Bytes) -> Result<()> {
-        Ok(self
-            .inner
+    async fn put(&self, location: &Path, bytes: Bytes) -> Result<PutResult> {
+        self.inner
             .write(location.as_ref(), bytes)
             .await
-            .map_err(|err| format_object_store_error(err, location.as_ref()))?)
+            .map_err(|err| format_object_store_error(err, location.as_ref()))?;
+        let meta = self
+            .inner
+            .stat(location.as_ref())
+            .await
+            .map_err(|err| format_object_store_error(err, location.as_ref()))?;
```

Comment:
Is there any way to avoid this extra `stat` call?

Reply:
It's possible for us to introduce the same thing. Before we implement it, we can set those field to `None`.

Reply:
Fixed https://github.com/apache/opendal/pull/4233/commits/877b72a2d7f99254660bc8e1c4732fa3c5fa35b9